src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationService.java

```diff
@@ -199,41 +200,43 @@ public Optional<String> validateMaximumSize(CaseData caseData) {
 
     public List<String> validateRespRepNames(CaseData caseData) {
         List<String> errors = new ArrayList<>();
-        if (CollectionUtils.isNotEmpty(caseData.getRespondentCollection())
-            && CollectionUtils.isNotEmpty(caseData.getRepCollection())) {
-            List<RepresentedTypeRItem> repCollection = caseData.getRepCollection();
-            List<RepresentedTypeRItem> updatedRepList = new ArrayList<>();
-            int repCollectionSize = caseData.getRepCollection().size();
-
-            //reverse update it - from the last to the first element by removing repetition
-            for (int index = repCollectionSize - 1;  index > -1; index--) {
-                String tempCollCurrentName = repCollection.get(index).getValue()
-                    .getDynamicRespRepName().getValue().getLabel();
-                if (isValidRespondentName(caseData, tempCollCurrentName)) {
-                    if (!repCollection.isEmpty()
-                        && updatedRepList.stream()
-                        .noneMatch(r -> r.getValue().getDynamicRespRepName().getValue().getLabel()
-                            .equals(tempCollCurrentName))) {
-                        repCollection.get(index).getValue().setRespRepName(tempCollCurrentName);
-                        updatedRepList.add(repCollection.get(index));
-                    }
-                } else {
-                    errors.add(RESP_REP_NAME_MISMATCH_ERROR_MESSAGE + " - " + tempCollCurrentName);
-                    return errors;
-                }
+        List<RepresentedTypeRItem> repCollection = caseData.getRepCollection();
```

Comment:
This is still a pretty huge method - any chance you could break it up into smaller chunks? Same goes for others. Happy to put a comment on the ticket so you can bump up the points